plugins/modules/dynamodb_table.py

```diff
@@ -174,300 +195,734 @@
+                "Index name: {0}".format(index['name']),
+                version='3.0.0', collection_name='community.aws')
+
+    idx = dict(
+        IndexName=index['name'],
+        KeySchema=key_schema,
+        Projection=projection,
+    )
+    if include_throughput:
+        idx['ProvisionedThroughput'] = throughput
 
+    return idx
 
-def dynamo_table_exists(table):
-    try:
-        table.describe()
-        return True
 
-    except JSONResponseError as e:
-        if e.message and e.message.startswith('Requested resource not found'):
-            return False
-        else:
-            raise e
+def _attribute_changes(current_table):
+    # TODO (future) It would be nice to catch attempts to change types here.
+    return _generate_attributes()
 
 
-def update_dynamo_table(table, throughput=None, check_mode=False, global_indexes=None):
-    table.describe()  # populate table details
-    throughput_changed = False
-    global_indexes_changed = False
-    if has_throughput_changed(table, throughput):
-        if not check_mode:
-            throughput_changed = table.update(throughput=throughput)
-        else:
-            throughput_changed = True
+def _global_index_changes(current_table):
+    current_global_index_map = current_table['_global_index_map']
+    global_index_map = _generate_global_index_map(current_table)
 
-    removed_indexes, added_indexes, index_throughput_changes = get_changed_global_indexes(table, global_indexes)
-    if removed_indexes:
-        if not check_mode:
-            for name, index in removed_indexes.items():
-                global_indexes_changed = table.delete_global_secondary_index(name) or global_indexes_changed
-        else:
-            global_indexes_changed = True
+    index_changes = list()
 
-    if added_indexes:
-        if not check_mode:
-            for name, index in added_indexes.items():
-                global_indexes_changed = table.create_global_secondary_index(global_index=index) or global_indexes_changed
+    # TODO (future) it would be nice to add support for deleting an index
+    for name in global_index_map:
+        idx = dict(_generate_index(global_index_map[name]))
+        if name not in current_global_index_map:
+            index_changes.append(dict(Create=idx))
         else:
-            global_indexes_changed = True
+            # The only thing we can change is the provisioned throughput.
+            # TODO (future) it would be nice to throw a deprecation here
+            # rather than dropping other changes on the floor
+            _current = current_global_index_map[name]
+            _new = global_index_map[name]
+            change = dict()
+            if _new['read_capacity'] != _current['read_capacity']:
+                change['ReadCapacityUnits'] = _new['read_capacity']
+            if _new['write_capacity'] != _current['write_capacity']:
+                change['WriteCapacityUnits'] = _new['write_capacity']
```

Comment:
I think this is omitting `ReadCapacityUnits` / `WriteCapacityUnits` when only one of them changes.
https://github.com/ansible-collections/community.aws/pull/726#issuecomment-924806413